hudi-timeline-service/src/main/java/org/apache/hudi/timeline/service/handlers/MarkerHandler.java

```diff
@@ -150,7 +168,50 @@ public boolean doesMarkerDirExist(String markerDir) {
    * @param markerName marker name
    * @return the {@code CompletableFuture} instance for the request
    */
-  public CompletableFuture<String> createMarker(Context context, String markerDir, String markerName) {
+  public CompletableFuture<String> createMarker(Context context, String markerDir, String markerName,
+                                                String batchInterval, String period, String maxAllowableHeartbeatIntervalInMs,
+                                                String basePath, String earlyConflictDetectionEnable,
+                                                String earlyConflictDetectionClassName) {
+    // Step1 do early conflict detection if enable
+    if (Boolean.parseBoolean(earlyConflictDetectionEnable)) {
+      try {
+        synchronized (earlyConflictDetectionLock) {
+          if (earlyConflictDetectionStrategy == null) {
+            earlyConflictDetectionStrategy = ReflectionUtils.loadClass(earlyConflictDetectionClassName);
+          }
+
+          if (!markerDir.equalsIgnoreCase(currentMarkerDir)) {
+            this.currentMarkerDir = markerDir;
+            Set<String> actions = CollectionUtils.createSet(COMMIT_ACTION, DELTA_COMMIT_ACTION, REPLACE_COMMIT_ACTION);
+            Set<HoodieInstant> oldInstants = viewManager.getFileSystemView(basePath)
+                .getTimeline()
+                .filterCompletedInstants()
+                .filter(instant -> actions.contains(instant.getAction()))
+                .getInstants()
+                .collect(Collectors.toSet());
+
+            earlyConflictDetectionStrategy.fresh(batchInterval, period, markerDir, basePath, maxAllowableHeartbeatIntervalInMs, fileSystem,
+                this, oldInstants);
+          }
+        }
+
+        if (earlyConflictDetectionStrategy.hasMarkerConflict()) {
+          earlyConflictDetectionStrategy.resolveMarkerConflict(basePath, markerDir, markerName);
```

Comment:
No exception should be thrown here at the timeline server if there is detected conflict.  The timeline server should simply return false for the marker creation request and let the executor/write handle resolve the marker conflict (throw the exception).

Reply:
Same reason, this check is batch and async. For specific request get false result. It means maker checker find a conflict but maybe it is not current request related marker conflict.

So is it possible to let current executor to handle others' conflict based on timeline sever in async and batch mode. :)